resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/InternalFileSystemCore.java

```diff
@@ -129,9 +130,9 @@ public IFileStore getStore(URI uri) throws CoreException {
 	 * Returns the fully initialized file system registry
 	 * @return The file system registry
 	 */
-	private synchronized HashMap<String, Object> getFileSystemRegistry() {
+	private synchronized Map<String, Object> getFileSystemRegistry() {
```

Comment:
If you want to make it clear

```suggestion
	private synchronized ConcurrentMap<String, Object> getFileSystemRegistry() {
```